include/solvers/auxiliary_physics.h

```diff
@@ -107,7 +107,7 @@ class AuxiliaryPhysics : public PhysicsSolver<VectorType>
    * To be defined for some physics only (eg. free surface, see vof.h).
    */
   virtual void
-  modify_solution(){};
+  modify_solution() = 0;
```

Comment:
Why make it an virtual interface? This means that every Auxiliary physics will need to implement an empty modify_solution(). It was ok like it was, no need to make it a pure virtual interface. It can have a default (empty) implementation.

Reply:
Corrected.